### PR TITLE
Inline replies

### LINKED ISSUE #1084: replies
new message type (19) https://discord.com/developers/docs/resources/channel#message-object-message-types

### PR DESCRIPTION
**Pull request by bsian03**
See #1027 
Inline replies are now live as per discord/discord-api-docs#2118

### CODE REVIEW

File: lib/rest/RequestHandler.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ class RequestHandler {
                     }
                     if(body && body.reason) { // Audit log reason sniping
                         headers["X-Audit-Log-Reason"] = body.reason;
-                        if(method !== "POST" || !url.includes("/prune")) {
+                        if((method !== "PUT" || !url.includes("/bans")) && (method !== "POST" || !url.includes("/prune"))) {
```
Comment (Khaaz) [inline lib/rest/RequestHandler.js]: looks like something was forgotten

File: lib/Client.js [hunk 2]
```diff
@@ -526,6 +526,10 @@ class Client extends EventEmitter {
     * @arg {Boolean | Array<String>} [content.allowedMentions.users] Whether or not to allow all user mentions, or an array of specific user mentions to allow.
     * @arg {String} content.content A content string
     * @arg {Object} [content.embed] An embed object. See [the official Discord API documentation entry](https://discord.com/developers/docs/resources/channel#embed-object) for object structure
+    * @arg {Object} [content.messageReference] An object containing reply data
+    * @arg {String} [content.messageReference.channelID] The channel ID of the message which should be replied to
```
Comment (Khaaz) [inline lib/Client.js]: seems like channelID and guildID are optional

**Review by Khaaz — COMMENTED** [review]
You also need to add alloweMentions.repliedUsers
https://github.com/discord/discord-api-docs/pull/2118/files#diff-160acb4654c611c67c4ee46b2ce5dda2be82212af8d1d00cd0256ef6edda1e95R564

File: lib/Constants.js [hunk 3]
```diff
@@ -180,7 +180,8 @@ module.exports.MessageTypes = {
     GUILD_DISCOVERY_DISQUALIFIED: 14,
     GUILD_DISCOVERY_REQUALIFIED: 15,
     GUILD_DISCOVERY_GRACE_PERIOD_INITIAL_WARNING: 16,
-    GUILD_DISCOVERY_GRACE_PERIOD_FINAL_WARNING: 17
+    GUILD_DISCOVERY_GRACE_PERIOD_FINAL_WARNING: 17,
+    INLINE_REPLY: 19
```
Comment (Khaaz) [inline lib/Constants.js]: This is named `REPLY` in discord side, should probably follow the same convention I guess

File: index.d.ts [hunk 4]
```diff
@@ -996,6 +1004,7 @@ declare namespace Eris {
       GUILD_DISCOVERY_REQUALIFIED: 15;
       GUILD_DISCOVERY_GRACE_PERIOD_INITIAL_WARNING: 16;
       GUILD_DISCOVERY_GRACE_PERIOD_FINAL_WARNING: 17;
+      INLINE_REPLY: 19;
```
Comment (AlekEagle) [inline index.d.ts]: same as https://github.com/abalabahaha/eris/pull/1069/files#r525348565

File: lib/rest/RequestHandler.js (hunk 1 above)
Comment (bsian03) [reply lib/rest/RequestHandler.js]: Why is this even in the PR????

**Review by Khaaz — DISMISSED** [review]

**Comment by AlekEagle** [comment]
Discord just unofficially rolled them out globally, the sooner the better
example

**Comment by bsian03** [comment]
1: 
image

2: The PR will be merged when abal merges it. There is also the option to just use my fork for the time being.

**Review by AlekEagle — DISMISSED** [review]

**Comment by bsian03** [comment]
It turns out that I didn't see the message that you can only create replies to the same channel you're replying to. Will fix this at some point later today

**Review by abalabahaha — APPROVED** [review]

File: index.d.ts [hunk 5]
```diff
@@ -684,6 +685,7 @@ declare namespace Eris {
     everyone?: boolean;
     roles?: boolean | string[];
     users?: boolean | string[];
+    replied_user?: boolean;
```
Comment (DetachHead) [inline index.d.ts:688]: the key name here is incorrect, it's actually `repliedUser`

Comment (miyacoz) [reply index.d.ts:688]: i made a fix for that a while ago #1099

**Pull request merged**